appcli/commands/service_cli.py

```diff
@@ -253,14 +275,17 @@ def _action_orchestrator(
         )
         post_run_log_message = f"{action.name} command finished with code [%i]"
 
-        logger.debug(f"Running pre-{action.name} hook")
-        pre_hook(ctx)
+        if pre_hook is not None:
+            logger.debug(f"Running pre-{action.name} hook")
+            pre_hook(ctx)
 
         logger.info(pre_run_log_message)
         result = action_run_function(ctx.obj, service_names)
 
-        logger.debug(f"Running post-{action.name} hook")
-        post_hook(ctx, result)
+        if post_hook is not None:
+            logger.debug(f"Running post-{action.name} hook")
+            post_hook(ctx, result)
 
+        click.echo(result.stdout)
```

Comment:
What's the purpose of this line, and do we need it?

There may be some implications of echoing to stdout. I'm not certain, but there might be issues with custom_commands that use stdout

Reply:
Actually I think the issue I'm thinking of is unfounded.
In some of our projects that use appcli, the `result` stdout is captured when we do `orchestrator.exec`, but here we're calling the click command which is outside that call stack, so I'm not concerned.